Shared/alerts/DynamicEntityLayerAlertTarget.cpp

```diff
@@ -0,0 +1,197 @@
+/*******************************************************************************
+ *  Copyright 2012-2024 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+// PCH header
+#include "pch.hpp"
+
+#include <QMap>
+
+#include "DynamicEntityLayerAlertTarget.h"
+
+// dsa app headers
+#include "GeometryQuadtree.h"
+#include "MessageFeed.h"
+#include "MessagesOverlay.h"
+
+// C++ API headers
+#include "DynamicEntity.h"
+#include "DynamicEntityInfo.h"
+#include "DynamicEntityLayer.h"
+#include "DynamicEntityObservation.h"
+#include "DynamicEntityObservationInfo.h"
+#include "Envelope.h"
+#include "Graphic.h"
+
+using namespace Esri::ArcGISRuntime;
+
+namespace Dsa {
+
+/*!
+  \class Dsa::DynamicEntityLayerAlertTarget
+  \inmodule Dsa
+  \inherits AlertTarget
+  \brief Represents a target based on an \l Esri::ArcGISRuntime::DynamicEntityLayer
+  for an \l AlertCondition.
+
+  Changes to any of the dynamic entities in the layer will cause the \l AlertTarget::locationChanged
+  signal to be emitted.
+  */
+
+
+/*!
+  \brief Constructor taking an \l Esri::ArcGISRuntime::DynamicEntityLayer (\a dynamicEntityLayer).
+
+  Entities are added/updated/removed by the DynamicEntityDataSource of the DynamicEntityLayer.
+ */
+DynamicEntityLayerAlertTarget::DynamicEntityLayerAlertTarget(DynamicEntityLayer* dynamicEntityLayer) :
+  AlertTarget(dynamicEntityLayer),
+  m_dynamicEntityLayer(dynamicEntityLayer),
+  m_entityGraphics(QMap<quint64, Graphic*>{})
+{
+  // subscribe to the entity received signal from the source of the dynamic layer
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityReceived, this, [this](DynamicEntityInfo* info)
+  {
+    // check the new entity for a connection and mark the info as delete later
+    auto* dynamicEntity = info->dynamicEntity();
+    connectEntityGraphic(dynamicEntity);
+    info->deleteLater();
+  });
+
+  // subscribe to the observation received signal (this essentially serves as the 'update' signal for the dynamic layer type)
+  connect(m_dynamicEntityLayer->dataSource(), &DynamicEntityDataSource::dynamicEntityObservationReceived, this, [this](DynamicEntityObservationInfo* observationInfo)
+  {
+    // check the entity for a connection and mark the observation as delete later
+    auto* dynamicEntity = observationInfo->observation()->dynamicEntity();
```

Comment:
ditto ownership/parent here. I think we should be explicit with our expectations of who owns this. We know it won't leak since we never return object without any parent, but in cases like this we need short lifetime for DynamicEntityObservationInfo, and longer lifetime for DynamicEntity.